### PR TITLE
core: dependency updates

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
Precursor work to the datagrid branch:
- Updated all Core dependencies and dev dependencies
- Updated storybook version and related issues]
- Updated and improve accuracy of the custom-elements.json metadata
- Added improved TS checks and linting problematic native property overrides
- Fix related modal hidden property override breaking conditional dom rendering APIs (only 6.x)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/build/eslint-rules/reserved-event-names.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return {
+      Decorator: node => {
+        const decoratorName = node.expression.callee.name;
+
+        if (decoratorName === 'event') {
+          const propName = node.parent.key.name;
+          if (isReservedEvent(propName)) {
+            context.report({
+              node: node.parent.key,
+              message: `"@${decoratorName} ${propName}" conflicts with a event in the HTMLElement base type. Please choose a different name.`,
```
Comment (coryrylan) [inline packages/core/build/eslint-rules/reserved-event-names.js:25]: This lint rule will help prevent accidental overrides of existing native events

File: packages/core/build/eslint-rules/reserved-property-names.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    return {
+      Decorator: node => {
+        const decoratorName = node.expression.callee.name;
+
+        if (decoratorName === 'property' || decoratorName === 'state' || decoratorName === 'event') {
+          const propName = node.parent.key.name;
+          if (isReservedProperty(propName)) {
+            context.report({
+              node: node.parent.key,
+              message: `"@${decoratorName} ${propName}" conflicts with a key in the HTMLElement prototype. Please choose a different name.`,
```
Comment (coryrylan) [inline packages/core/build/eslint-rules/reserved-property-names.js:24]: This lint rule will help prevent accidental overrides of existing native properties

File: packages/core/.storybook/public/preview.css [hunk 3]
```diff
@@ -1,10 +1,10 @@
 @import '@cds/city/css/bundles/default.min.css';
```
Comment (coryrylan) [inline packages/core/.storybook/public/preview.css:1]: Storybook APIs changed a bit. This file was cleaned up to relfect the APIs. There is a css file for the preview/iframe and a new file for the css only to be applied to the manager UI. Any commonly shared styles are now in `common.css`

File: packages/core/src/card/card.stories.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ export function WithLayout() {
 }
 
 /** @website **/
-export function WithContainerOfCards() {
+export function withContainerOfCards() {
```
Comment (coryrylan) [inline packages/core/src/card/card.stories.ts:54]: Newer version enforces case consistency with story names

File: packages/core/rollup.config.js [hunk 5]
```diff
@@ -98,7 +98,16 @@ export default [
...
+      prod
+        ? replace({
+            preventAssignment: false,
+            values: {
+              PACKAGE_VERSION: version,
+              'super(...arguments),': 'super(...arguments);',
+              'super(),': 'super();',
+            }, // safari 15.1 bug with minification optimization
+          })
+        : [],
```
Comment (coryrylan) [inline packages/core/rollup.config.js:110]: This hopefully is a temporary issue. Once Safari releases a new version that does not fail on the valid syntax used by the minifier we can remove this.

File: packages/core/stylelint.config.js [hunk 6]
```diff
@@ -37,9 +39,9 @@ const rules = {
 const config = {
   extends: 'stylelint-config-recommended',
   defaultSeverity: 'error',
-  syntax: 'scss',
+  customSyntax: postcssScss,
   plugins: ['stylelint-scss', 'stylelint-declaration-use-variable', 'stylelint-no-px'],
-  ignoreFiles: ['./src/styles/tokens/generated/*.scss', './dist/**/*.scss'],
+  ignoreFiles: ['./src/styles/tokens/generated/*.scss', './src/styles/shim.clr-ui.scss', './dist/**/*.scss'],
   rules,
```
Comment (coryrylan) [inline packages/core/stylelint.config.js:45]: minor api updates with PostCSS

File: packages/core/web-dev-server.config.mjs [hunk 7]
```diff
@@ -6,13 +6,16 @@
...
 /** Use Hot Module replacement by adding --hmr to the start command */
 const hmr = process.argv.includes('--hmr');
 
 export default /** @type {import('@web/dev-server').DevServerConfig} */ ({
-  nodeResolve: true,
   open: './',
   watch: !hmr,
+  nodeResolve: {
+    exportConditions: ['development'],
+  },
```
Comment (coryrylan) [inline packages/core/web-dev-server.config.mjs:18]: The `exportConditions` allows libraries like lit to provide a dev build to catch more errors during development. When building these libraries will default to prod optimized builds.

File: packages/core/web-test-runner.a11y.mjs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  return {
+    name: 'a11y',
+    async executeCommand({ command, payload, session }) {
+      if (command === 'a11y:snapshot' && session.browser.type === 'playwright') {
+        const page = session.browser.getPage(session.id);
+        const root = await page.$(payload.root);
+        const snapshot = await page.accessibility.snapshot({ interestingOnly: false, root });
+        return snapshot;
+      }
+    },
```
Comment (coryrylan) [inline packages/core/web-test-runner.a11y.mjs:51]: The is precursor work to datagrid branch. Trying to break it up however no tests are in this PR and will be added in follow up

File: packages/core/.storybook/public/preview.css [hunk 9]
```diff
@@ -106,33 +102,15 @@ iframe[resizable] {
...
-  background: transparent !important;
-}
-
 .docs-story div {
   /* forcing storybook to not add transforms on their story blocks. it was messing up modal/overlay/dropdown demos */
   transform: none !important;
 }
 
+pre.prismjs {
+  background: hsl(0, 0%, 98%) !important;
```
Comment (hippee-lee) [inline packages/core/.storybook/public/preview.css]: Could / should we use the `--cds-global-color-gray-50` token here?

File: packages/core/tsconfig.json [hunk 10]
```diff
@@ -1,11 +1,11 @@
 {
   "compilerOptions": {
-    "target": "es2019",
+    "target": "esnext",
```
Comment (hippee-lee) [inline packages/core/tsconfig.json]: why `esnext` and not `es2021`? Isn't `esnext` [unpredictable](https://www.typescriptlang.org/tsconfig#target)?

File: packages/core/tsconfig.storybook.json [hunk 11]
```diff
@@ -1,17 +1,35 @@
...
+    "./**/*.spec.ts",
+    "./**/*.a11y.ts",
+    "./**/*.performance.ts",
+    "./build/**/*.ts"
+  ],
   "compilerOptions": {
-    "rootDirs": ["src", "docs"],
-    "target": "es2017",
+    "rootDir": "./",
+    "target": "esnext",
```
Comment (hippee-lee) [inline packages/core/tsconfig.storybook.json]: Same comment as above

**Review by hippee-lee — APPROVED** [review]
I had a question re: the best target value esnext vs es2021. Not a blocker.

File: packages/core/tsconfig.json (hunk 10 above)
Comment (coryrylan) [reply packages/core/tsconfig.json]: Yeah, agree it should be 2021. Not sure why I had next. Will fixup

File: packages/core/tsconfig.storybook.json (hunk 11 above)
Comment (coryrylan) [reply packages/core/tsconfig.storybook.json]: Agree, will fixup

File: packages/core/build/eslint-rules/reserved-property-names.js [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+  'scrollIntoViewIfNeeded',
+  'animate',
+  'computedStyleMap',
+  'before',
+  'after',
+  'replaceWith',
+  'remove',
+  'prepend',
+  'append',
+  'querySelector',
```
Comment (mathisscott) [inline packages/core/build/eslint-rules/reserved-property-names.js:268]: What about the aria attributes and properties? Are we covering those too?

**Review by mathisscott — APPROVED** [review]

File: packages/core/build/eslint-rules/reserved-property-names.js (hunk 12 above)
Comment (coryrylan) [reply packages/core/build/eslint-rules/reserved-property-names.js:268]: Ah thanks! That was the entire reason I started digging into this and forgot to add them 😅 I'll get it patched up

**Pull request merged**